packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/BarCodeScannerView.kt

```diff
@@ -2,39 +2,41 @@ package expo.modules.barcodescanner
 
 import android.content.Context
 import android.hardware.SensorManager
+import android.os.Bundle
+import android.util.Pair
 import android.view.OrientationEventListener
 import android.view.View
-import android.view.ViewGroup
 import android.view.WindowManager
-import expo.modules.barcodescanner.BarCodeScannedEvent.Companion.obtain
+import androidx.core.util.component1
+import androidx.core.util.component2
 import expo.modules.barcodescanner.utils.mapX
 import expo.modules.barcodescanner.utils.mapY
-import expo.modules.core.ModuleRegistryDelegate
-import expo.modules.core.interfaces.services.EventEmitter
 import expo.modules.interfaces.barcodescanner.BarCodeScannerResult
 import expo.modules.interfaces.barcodescanner.BarCodeScannerSettings
+import expo.modules.kotlin.AppContext
+import expo.modules.kotlin.viewevent.EventDispatcher
+import expo.modules.kotlin.views.ExpoView
 import kotlin.math.roundToInt
 
 class BarCodeScannerView(
-  private val viewContext: Context,
-  private val moduleRegistryDelegate: ModuleRegistryDelegate
-) : ViewGroup(viewContext) {
+  context: Context,
+  appContext: AppContext,
+) : ExpoView(context, appContext) {
+  private val onBarCodeScanned by EventDispatcher<BarCodeScannedEvent>()
```

Comment:
The previous version used the coalescing key to reduce the number of events sent to the js. You can use it as well in the new API. Here is an example: https://github.com/expo/expo/blob/9a8b890192f15de7ae98743b6c387d2a8847ddd5/packages/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt#L74.